sled-agent/src/bootstrap/agent.rs

```diff
@@ -733,6 +738,9 @@ impl Agent {
         &self,
         request: RackInitializeRequest,
     ) -> Result<(), BootstrapError> {
+        // Avoid concurrent initialization and teardown.
+        let _rss_access = self.rss_access.lock().await;
```

Comment:
Should these be `try_lock()` and return an error immediately if the lock is held? Otherwise the caller is going to be waiting a while for a response, I think, and perhaps they don't even really want this to attempt to run twice.

Reply:
I like that idea; done in bdf912723bf151d2f1f7a06cc938f5990326b3e0